samples/winapi_drivelist/main.c

```diff
@@ -9,45 +9,46 @@ int main(void)
 
     // Mount some drives for demonstration purposes
     BOOL ret;
+    DWORD error;
     ret = nxMountDrive('C', "\\Device\\Harddisk0\\Partition2\\");
     if (!ret) {
-        debugPrint("Failed to mount C: drive!\n");
-        Sleep(5000);
-        return 1;
+        // Additional error info can be retrieved with GetLastError()
+        error = GetLastError();
+        debugPrint("Failed to mount C: drive! Error code: %x\n", error);
+        goto end;
     }
+
     ret = nxMountDrive('E', "\\Device\\Harddisk0\\Partition1\\");
     if (!ret) {
-        debugPrint("Failed to mount E: drive!\n");
-        Sleep(5000);
-        return 1;
+        error = GetLastError();
+        debugPrint("Failed to mount E: drive! Error code: %x\n", error);
+        goto unmount_c;
     }
 
-    // Retrieve drive bitmaks. Every bit represents one drive letter
+    // Retrieve drive bitmasks. Every bit represents one drive letter
     DWORD driveBits = GetLogicalDrives();
-    if (driveBits == 0 && GetLastError() != ERROR_SUCCESS) {
-        debugPrint("Failed to retrieve drive bitmask!\n");
-        Sleep(5000);
-        return 1;
+    error = GetLastError();
+    if (driveBits == 0 && error != ERROR_SUCCESS) {
+        debugPrint("Failed to retrieve drive bitmask! Error code: %x\n", error);
+        goto unmount_e;
```

Comment:
Here you jump to `unmount_e`, but in https://github.com/XboxDev/nxdk/pull/421/commits/f9e1ed24b6617932311ba1a90e216bde05eadc25#r611092579 you jump to `unmount_c`

Both of these jump targets are confusing.

Reply:
I believe that with the change to `cleanup` this will make more sense.